checkout/frames-android
# PIMOB: 2117 - adding capabilities for phone number and country field optional and hidden

## Issue

[PIMOB-2117](https://checkout.atlassian.net/browse/PIMOB-2117)

## Proposed changes

- Adding capabilities for making phone number and country fields optional and hidden
- Tokenisation logic changes accordingly. Explaining it, if any fields are set as optional or hidden and the object for country and phone is null or empty, then a token can be created successfully.

Scenarios in data models
1. If the phone number is optional and not filled then the phone object should not pass in tokenisation
2. If the country is optional and not selected, then the country field in tokenisation should not pass
3. if the phone number is mandatory and the country field is optional, then if the country is not selected and the phone number is entered then the phone number object only number should pass.


## Test Steps
1. Add country as an optional field in the custom sample app, then verify without selecting country user is able to save the billing form and accordingly reflected in the payment details form
2. Add phone as an optional field in the custom sample app, then verify without entering phone user is able to save the billing form and accordingly reflected in the payment details form
3. Add country and phone both optional then without entering and selecting country user should be able to save the billing form and accordingly reflected in the payment details form
In above all scenarios, user should be able to create a token

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._


## Further comments

_If this is a relatively large or complex change, kick off the discussion by explaining why you choose the solution you did and what alternatives you considered, etc..._


[PIMOB-2117]: https://checkout.atlassian.net/browse/PIMOB-2117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
